File: libs/as-sdk-integration-tests/assembly/proxy-http.ts
import {
	Bytes,
	OracleProgram,
	Process,
	generateProxyHttpSigningMessage,
	proxyHttpFetch,
} from "../../as-sdk/assembly";

export class TestProxyHttpFetch extends OracleProgram {
	execution(): void {
		const response = proxyHttpFetch("http://localhost:5384/proxy/planets/1");

		if (response.ok) {
			Process.success(response.bytes);
		}

		Process.error(response.bytes);
	}
}

export class TestGenerateProxyMessage extends OracleProgram {
	execution(): void {
		const message = generateProxyHttpSigningMessage(
			"https://example.com",
			"get",
			Bytes.empty(),
			Bytes.fromUtf8String(`{"name":"data-proxy"}`),
		);

		Process.success(Bytes.fromUtf8String(message.toHexString()));
	}
}

File: libs/go-sdk-integration-tests/tests/proxy-http.go
package tests

import (
	"encoding/hex"

	"github.com/seda-protocol/seda-sdk/http"
	"github.com/seda-protocol/seda-sdk/process"
)

func TestProxyHttpFetch() {
	response := http.ProxyHttpFetch("http://localhost:5384/proxy/planets/1", nil, nil)

	if response.IsOk() {
		process.Success(response.Bytes)
	}

	process.Error(response.Bytes)
}

func TestGenerateProxyMessage() {
	response := http.GenerateProxyHttpSigningMessage("https://example.com", "GET", nil, []byte("{\"name\":\"data-proxy\"}"))

	process.Success([]byte(hex.EncodeToString(response)))
}
